### ISSUE TITLE
read-function of plugin `thermal' failed.

### ISSUE THREAD

**Issue by github-jeff** [collectd/collectd#1972]
*   Version of collectd:
collectd 5.4.0.git, http://collectd.org/

*   Operating system / distribution:
Ubuntu 14.04 Server - Up to date on all packages and kernels

## Expected behavior

/etc/collectd/collectd.conf

```
LoadPlugin thermal
<Plugin thermal>
        ForceUseProcfs false
        Device "temp1"
        Device "temp2"
        Device "Core 0"
        Device "Core 1"
        Device "acpitz-virtual-0"
        Device "coretemp-isa-0000"
        IgnoreSelected false
</Plugin>
```

Expected results from sensors:

```
#sensors
acpitz-virtual-0
Adapter: Virtual device
temp1:        +27.8°C  (crit = +105.0°C)
temp2:        +29.8°C  (crit = +105.0°C)

coretemp-isa-0000
Adapter: ISA adapter
Physical id 0:  +36.0°C  (high = +80.0°C, crit = +100.0°C)
Core 0:         +34.0°C  (high = +80.0°C, crit = +100.0°C)
Core 1:         +29.0°C  (high = +80.0°C, crit = +100.0°C)
```

## Actual behavior

/var/log/syslog

```
Exiting normally.
collectd[7631]: collectd: Stopping 5 read threads.
collectd[7631]: ping plugin: Shutting down thread.
collectd[7631]: rrdtool plugin: Shutting down the queue thread.
 collectd[7631]: collectd: Stopping 5 write threads.
collectdmon[7630]: Info: collectd terminated with exit status 0
collectdmon[7630]: Info: shutting down collectdmon
collectd[7748]: cpufreq plugin: Found 2 CPUs
collectd[7748]: Initialization complete, entering read-loop.
collectd[7748]: rrdtool plugin: Adjusting "RandomTimeout" to 0.000 seconds.
collectd[7748]: read-function of plugin `thermal' failed. Will suspend it for 20.000 seconds.
collectd[7748]: read-function of plugin `thermal' failed. Will suspend it for 40.000 seconds.
collectd[7748]: read-function of plugin `thermal' failed. Will suspend it for 80.000 seconds.
collectd[7748]: read-function of plugin `thermal' failed. Will suspend it for 160.000 seconds.
collectd[7748]: read-function of plugin `thermal' failed. Will suspend it for 320.000 seconds.
```

And nothing is reported to graphite.

**Comment by github-jeff** [comment]
Works better when no Device "  " is specified.  Thermal just sends everything over and you can pick what you want monitored downstream.

**Comment by octo** [comment]
I have a feeling that the *thermal plugin* is not using the `walk_directory()` utility function correctly: it returns -1 when a device matches the "ignore list", causing the directory walk to abort and return an error. Instead, it should continue with the next entry in the directory.

**Comment by octo** [comment]
@github-jeff one of the ignorelist is set up in a way to ignore *all* available devices. Can you please check that there is at least one device left to read?

Best regards,
—octo

**Comment by github-jeff** [comment]
When not initially specifying a device all devices available were reported instead, which is perfectly fine.  I just cherry picked the devices that I wanted (i.e. had reported data) which ended up being:

thermal-thermal_zone0
thermal-thermal_zone1
thermal-thermal_zone2

```
<Plugin thermal>
      ForceUseProcfs false
      IgnoreSelected false
</Plugin>
```

Other Devices reported were thermal-cooling_device0 thru device9.  None of these had any reported data.

![image](https://cloud.githubusercontent.com/assets/20625700/19022362/cc043814-88a4-11e6-8f93-275a2acbd2ea.png)

When I went back to try and specify the Device ="thermal-thermal_zone0" thermal module failed again.  So I just leave the plugin section as shown and filter out what I don't need in graphite.

I did not see anything about an ignore list.  Where is it located?
